Foundation/src/URI.cpp

```diff
@@ -720,8 +720,8 @@ unsigned short URI::getWellKnownPort() const
 void URI::parse(const std::string& uri)
 {
 	std::for_each(uri.begin(), uri.end(), [] (char ch) {
-		if (static_cast<signed char>(ch) <= 32 || ch == '\x7F')
-			throw URISyntaxException("URI contains invalid characters");
+		if ((!std::isspace(ch) && static_cast<signed char>(ch) <= 32) || (ch == '\x7F'))
```

Comment:
But URL with whitespace is invalid. And URL with tab or newline is also invalid and dangerous (if it is pasted to HTTP request, we will get "request splitting" attack).

Why don't encode it before calling Poco::URI constructor?

Reply:
The names of hdfs files are in url and if I encode them beforehand, they come into hdfs undecoded: `error: File does not exist: /test%2Ftest%20test%20test%201%2Etxt` (because looks like it is then encoded twice and decoded once).

Reply:
Maybe it's a bug in `libhdfs3`? And it can be patched.
I'm afraid changing Poco::URI for this issue.

Reply:
Ok, I'll take a look at `libhdfs3`.